dashboard-ui/src/components/Survey/survey.jsx

```diff
@@ -1,48 +1,55 @@
-import React from "react";
+import React, { Component } from "react";
 import 'survey-core/defaultV2.min.css'
 import { Model } from 'survey-core'
 import { Survey } from "survey-react-ui"
 import surveyConfig from './surveyConfig.json';
 import surveyTheme from './surveyTheme.json';
 import gql from "graphql-tag";
 import { Mutation } from '@apollo/react-components';
+import { StaticTemplate } from "./staticTemplate";
+import { DynamicTemplate } from "./dynamicTemplate";
+import { getUID, shuffle } from './util'
 
 const UPLOAD_SURVEY_RESULTS = gql`
-  mutation UploadSurveyResults($results: JSON) {
-    uploadSurveyResults(results: $results)
+  mutation UploadSurveyResults( $surveyId: String, $results: JSON) {
+    uploadSurveyResults(surveyId: $surveyId, results: $results)
   }`;
 
-class SurveyPage extends React.Component {
+class SurveyPage extends Component {
 
     constructor(props) {
         super(props);
-        this.state = { uploadData: false };
+        this.state = {
+            uploadData: false,
+            startTime: null,
+            firstPageCompleted: false,
+            surveyId: null,
+            surveyVersion: surveyConfig.version,
+            iPad: false
+        };
+
         this.initializeSurvey();
         this.survey = new Model(surveyConfig);
         this.survey.applyTheme(surveyTheme)
         this.pageStartTimes = {};
         this.surveyData = {};
         this.survey.onAfterRenderPage.add(this.onAfterRenderPage);
+        this.survey.onValueChanged.add(this.onValueChanged)
         this.survey.onComplete.add(this.onSurveyComplete);
         this.uploadButtonRef = React.createRef();
     }
 
-    shuffle(array) {
-        // randomize the list
-        for (let i = array.length - 1; i > 0; i--) {
-            const j = Math.floor(Math.random() * (i + 1));
-            [array[i], array[j]] = [array[j], array[i]];
-        }
-        return array;
-    }
-
-    configureSurveyPages(groupedDMs, comparisonPages) {
-        /*
-        * There are three groups of decision makers, each group has two decision makers
-        * The order in which the groups gets presented is randomized
-        * The order of the decision makers within the groups is randomized
-        */
+    configureSurveyPages = (groupedDMs, comparisonPages) => {
+        // set pages to dynamic or static
+        /* commenting out randomization of dynamic or static for 2-2
+        Object.entries(templateAssignment).forEach(([pageName, templateType]) => {
+            const page = surveyConfig.pages.find(page => page.name === pageName);
+            if (page) {
+                page.elements[0].type = `${templateType}-template`;
+            }
+        });*/
```

Comment:
If this is something that will often be turned on or off, maybe add a .env variable to configure it instead of commenting on and off every time

Reply:
this is a temporary measure from what Ewart emailed me yesterday. My understanding is that normally this will not be turned off

Reply:
Awesome, just wanted to check :)